### PR TITLE
feat: add Sun and Moon gravitational models 

### PR DESCRIPTION
**Pull request by apaletta3**

### CODE REVIEW

**Review by apaletta3 — COMMENTED** [review]
Nice changes! I like the added clarity in the python bindings. LGTM and ready to merge

**Review by Derollez — APPROVED** [review]
LGTM!
- @apaletta3 Do you confirm all Jupyter make commands are running successfully?

File: bindings/python/src/OpenSpaceToolkitPhysicsPy/Environment/Gravitational.cpp [hunk 1]
```diff
@@ -24,6 +26,10 @@ inline void                     OpenSpaceToolkitPhysicsPy_Environment_Gravitatio
     // Add objects to python "gravitational" submodules
     OpenSpaceToolkitPhysicsPy_Environment_Gravitational_Spherical(gravitational) ;
     OpenSpaceToolkitPhysicsPy_Environment_Gravitational_Earth(gravitational) ;
+    OpenSpaceToolkitPhysicsPy_Environment_Gravitational_Sun(gravitational) ;
+    OpenSpaceToolkitPhysicsPy_Environment_Gravitational_Moon(gravitational) ;
+
+
 }
```
Comment (robinpdm) [inline bindings/python/src/OpenSpaceToolkitPhysicsPy/Environment/Gravitational.cpp:33]:
```suggestion
}
```
extra newline?

File: bindings/python/src/OpenSpaceToolkitPhysicsPy/Environment/Objects/CelestialBodies/Moon.cpp [hunk 2]
```diff
@@ -37,7 +44,7 @@ inline void                     OpenSpaceToolkitPhysicsPy_Environment_Objects_Ce
             .def("__repr__", &(shiftToString<Moon>))
 
             .def_static("default", &Moon::Default)
-
+            .def_static("spherical", &Moon::Spherical)
         ;
```
Comment (robinpdm) [inline bindings/python/src/OpenSpaceToolkitPhysicsPy/Environment/Objects/CelestialBodies/Moon.cpp:48]:
```suggestion
            .def_static("spherical", &Moon::Spherical)

        ;
```

File: include/OpenSpaceToolkit/Physics/Environment/Gravitational/Moon.hpp [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+using ostk::physics::units::Derived ;
+using ostk::physics::time::Instant ;
+using ostk::physics::environment::gravitational::Model ;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// @brief                      Moon gravitational model
+///
+///                             The gravitational potential of the moon for now is kept as a simple spherical model.
```
Comment (robinpdm) [inline include/OpenSpaceToolkit/Physics/Environment/Gravitational/Moon.hpp:51]:
```suggestion
///                             The gravitational potential of the Moon for now is kept as a simple spherical model.
```

File: include/OpenSpaceToolkit/Physics/Environment/Gravitational/Moon.hpp [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        ///
+        /// @param              [in] aType A gravitational model type
+        /// @param              [in] (optional) aDataDirectory A gravitational model data directory
+
+                                Moon                                        (   const   Moon::Type&                 aType,
+                                                                                const   Directory&                  aDataDirectory                              =   Directory::Undefined()  ) ;
+
+        /// @brief              Copy constructor
+        ///
+        /// @param              [in] aMoonGravitationalModel An Moon model
```
Comment (robinpdm) [inline include/OpenSpaceToolkit/Physics/Environment/Gravitational/Moon.hpp:73]:
```suggestion
        /// @param              [in] aMoonGravitationalModel A Moon model
```

File: include/OpenSpaceToolkit/Physics/Environment/Gravitational/Moon.hpp [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+                                Moon                                        (   const   Moon&                       aMoonGravitationalModel                     ) ;
+
+        /// @brief              Copy assignment operator
+        ///
+        /// @param              [in] aMoonGravitationalModel An Moon model
```
Comment (robinpdm) [inline include/OpenSpaceToolkit/Physics/Environment/Gravitational/Moon.hpp:79]:
```suggestion
        /// @param              [in] aMoonGravitationalModel A Moon model
```

File: include/OpenSpaceToolkit/Physics/Environment/Gravitational/Sun.hpp [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+using ostk::physics::units::Derived ;
+using ostk::physics::time::Instant ;
+using ostk::physics::environment::gravitational::Model ;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// @brief                      Sun gravitational model
+///
+///                             The gravitational potential of the sun for now is kept as a simple spherical model.
```
Comment (robinpdm) [inline include/OpenSpaceToolkit/Physics/Environment/Gravitational/Sun.hpp:51]:
```suggestion
///                             The gravitational potential of the Sun for now is kept as a simple spherical model.
```

File: include/OpenSpaceToolkit/Physics/Environment/Gravitational/Sun.hpp [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        ///
+        /// @param              [in] aType A gravitational model type
+        /// @param              [in] (optional) aDataDirectory A gravitational model data directory
+
+                                Sun                                         (   const   Sun::Type&                  aType,
+                                                                                const   Directory&                  aDataDirectory                              =   Directory::Undefined()  ) ;
+
+        /// @brief              Copy constructor
+        ///
+        /// @param              [in] aSunGravitationalModel An Sun model
```
Comment (robinpdm) [inline include/OpenSpaceToolkit/Physics/Environment/Gravitational/Sun.hpp:73]:
```suggestion
        /// @param              [in] aSunGravitationalModel A Sun model
```

File: include/OpenSpaceToolkit/Physics/Environment/Gravitational/Sun.hpp [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+
+                                Sun                                         (   const   Sun&                        aSunGravitationalModel                      ) ;
+
+        /// @brief              Copy assignment operator
+        ///
+        /// @param              [in] aSunGravitationalModel An Sun model
```
Comment (robinpdm) [inline include/OpenSpaceToolkit/Physics/Environment/Gravitational/Sun.hpp:79]:
```suggestion
        /// @param              [in] aSunGravitationalModel A Sun model
```

File: bindings/python/src/OpenSpaceToolkitPhysicsPy/Environment/Gravitational.cpp [hunk 9]
```diff
@@ -24,6 +26,9 @@ inline void                     OpenSpaceToolkitPhysicsPy_Environment_Gravitatio
     // Add objects to python "gravitational" submodules
     OpenSpaceToolkitPhysicsPy_Environment_Gravitational_Spherical(gravitational) ;
     OpenSpaceToolkitPhysicsPy_Environment_Gravitational_Earth(gravitational) ;
+    OpenSpaceToolkitPhysicsPy_Environment_Gravitational_Sun(gravitational) ;
+    OpenSpaceToolkitPhysicsPy_Environment_Gravitational_Moon(gravitational) ;
```
Comment (robinpdm) [inline bindings/python/src/OpenSpaceToolkitPhysicsPy/Environment/Gravitational.cpp:30]: To keep a consistent ordering:
```suggestion
    OpenSpaceToolkitPhysicsPy_Environment_Gravitational_Spherical(gravitational) ;
    OpenSpaceToolkitPhysicsPy_Environment_Gravitational_Sun(gravitational) ;
    OpenSpaceToolkitPhysicsPy_Environment_Gravitational_Moon(gravitational) ;
    OpenSpaceToolkitPhysicsPy_Environment_Gravitational_Earth(gravitational) ;
```

**Review by robinpdm — APPROVED** [review]

**Pull request merged**